src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -246,26 +245,28 @@ public Task Get()
             TransactionOperationContext context;
             using (ServerStore.ContextPool.AllocateOperationContext(out context))
             {
-                var updateJson = await context.ReadForMemoryAsync(RequestBodyStream(), "read-topology-update");
-                context.OpenReadTransaction();
-                long etag;
-                var databaseRecord = ServerStore.Cluster.ReadDatabase(context, name, out etag);
-                var index = await ServerStore.UpdateDatabaseTopology(context, name, updateJson);
-                await ServerStore.Cluster.WaitForIndexNotification(index);
+                var updateJson = await context.ReadForMemoryAsync(RequestBodyStream(), "read-modify-watchers");          
```

Comment:
Well, here I can protect it by doing something like that :
```
if (updateJson.TryGet("NewWatchers", out BlittableJsonReaderArray watchers) == false)
{
          throw new InvalidDataException("NewWatchers property was not found.");
} 
```
and pass the watchers further.
basically I try to avoid here from converting from blittable to entity, back to blittable and again to entity when applying the command.